src/topology/schema.rs

```diff
@@ -313,125 +347,126 @@ mod tests {
-                    inputs: vec![("source-foo", None)],
-                    sources: IndexMap::from([(
-                        "source-foo",
-                        vec![Output::default(DataType::all()).with_schema_definition(
-                            Definition::empty().with_field(
-                                "foo",
-                                Kind::integer().or_bytes(),
-                                Some("foo bar"),
-                            ),
-                        )],
-                    )]),
-                    transforms: IndexMap::default(),
-                    want: Definition::empty().with_field(
-                        "foo",
-                        Kind::integer().or_bytes(),
-                        Some("foo bar"),
-                    ),
-                },
-            ),
-            (
-                "multiple inputs, sources with schema",
-                TestCase {
-                    inputs: vec![("source-foo", None), ("source-bar", None)],
-                    sources: IndexMap::from([
-                        (
-                            "source-foo",
-                            vec![Output::default(DataType::all()).with_schema_definition(
-                                Definition::empty().with_field(
-                                    "foo",
-                                    Kind::integer().or_bytes(),
-                                    Some("foo bar"),
-                                ),
-                            )],
-                        ),
-                        (
-                            "source-bar",
-                            vec![Output::default(DataType::all()).with_schema_definition(
-                                Definition::empty().with_field(
-                                    "foo",
-                                    Kind::timestamp(),
-                                    Some("baz qux"),
-                                ),
-                            )],
-                        ),
-                    ]),
-                    transforms: IndexMap::default(),
-                    want: Definition::empty()
-                        .with_field(
-                            "foo",
-                            Kind::integer().or_bytes().or_timestamp(),
-                            Some("foo bar"),
-                        )
-                        .with_known_meaning("foo", "baz qux"),
-                },
-            ),
-        ]) {
-            let inputs = case
-                .inputs
-                .iter()
-                .cloned()
-                .map(|(key, port)| OutputId {
-                    component: key.into(),
-                    port,
-                })
-                .collect::<Vec<_>>();
-
-            let got = merged_definition(&inputs, &case, &mut HashMap::default());
-            assert_eq!(got, case.want, "{}", title);
-        }
-    }
+    // this is just testing transforms with no inputs... which isn't valid?
```

Comment:
I can't remember, but I assume the transforms without inputs is because it doesn't matter for the case under test that those have no inputs?

Reply:
ah, yea. This is related to the "TODO" I have above. I'll fix this, it just got lost.